src/platform/silabs/SilabsConfig.cpp

```diff
@@ -284,6 +284,11 @@ CHIP_ERROR SilabsConfig::WriteConfigValueStr(Key key, const char * str, size_t s
 
     if (str != NULL)
     {
+        // Compute the length of the string if not provided.
+        if (strLen == 0)
+        {
+            strLen = strlen(str);
```

Comment:
If str is not null-terminated, then this will cause out of bound read and write garbage to nvm3 possibly a large amount of data.

also, not to important but if the user did provide a strLen of 0 and it isn't the default value, then we are doing strlen for no reason but that not to dramatic.

Reply:
You are correct, however the unit test are expecting a callable API without a size. I think we could configure a maximum size nvm3 object, but in the event someone writes a non-null terminated string in a buffer that is smaller than that max, we would still have the possibility for overflows.

Reply:
can it be possible it's a null string with just a key?

Reply:
we have a check on NULL 5 lines above this line.